### PR TITLE
Polish Dragonwell

### PR DESCRIPTION
**Pull request by joeyleeeeeee97**
Related: 
https://github.com/AdoptOpenJDK/openjdk-build/issues/2385
https://github.com/AdoptOpenJDK/openjdk-build/issues/2316

### CODE REVIEW

File: sbin/build.sh [hunk 1]
```diff
@@ -149,11 +149,25 @@ getOpenJdkVersion() {
...
       version="jdk-${corrVersion}.${minorNum}.${updateNum}+${buildNum}.${fixNum}"
     fi
-  else
-    version=${BUILD_CONFIG[TAG]:-$(getFirstTagFromOpenJDKGitRepo)}
-    if [ "${BUILD_CONFIG[BUILD_VARIANT]}" == "${BUILD_VARIANT_DRAGONWELL}" ]; then
+  elif [ "${BUILD_CONFIG[BUILD_VARIANT]}" == "${BUILD_VARIANT_DRAGONWELL}" ]; then
+    local dragonwellVerFile=${BUILD_CONFIG[WORKSPACE_DIR]}/${BUILD_CONFIG[WORKING_DIR]}/${BUILD_CONFIG[OPENJDK_SOURCE_DIR]}/version.txt
+    if [ -r "${dragonwellVerFile}" ]; then
+      if [ "${BUILD_CONFIG[OPENJDK_CORE_VERSION]}" == "${JDK8_CORE_VERSION}" ]; then
+        local updateNum="$(cut -d'.' -f 2 <${corrVerFile})"
```
[the remark this answers is not in the document]
Comment (joeyleeeeeee97) [reply sbin/build.sh:156]: should be dragonwellVerFile, fixed.

File: sbin/build.sh [hunk 2]
```diff
@@ -819,14 +833,17 @@ getFirstTagFromOpenJDKGitRepo() {
     get_tag_cmd=$jdk8_get_tag_cmd
   fi
 
+  if [ "${BUILD_CONFIG[BUILD_VARIANT]}" == "${BUILD_VARIANT_DRAGONWELL}" ]; then
+    TAG_SEARCH="dragonwell-*_jdk*"
```
[the remark this answers is not in the document]
Comment (joeyleeeeeee97) [reply sbin/build.sh:837]: https://github.com/alibaba/dragonwell8/tags
https://github.com/alibaba/dragonwell11/tags

Yes. I think now current tags met this requirement.

File: sbin/build.sh [hunk 3]
```diff
@@ -149,11 +149,25 @@ getOpenJdkVersion() {
       local fixNum="$(cut -d'.' -f 5 <${corrVerFile})"
       version="jdk-${corrVersion}.${minorNum}.${updateNum}+${buildNum}.${fixNum}"
     fi
-  else
-    version=${BUILD_CONFIG[TAG]:-$(getFirstTagFromOpenJDKGitRepo)}
-    if [ "${BUILD_CONFIG[BUILD_VARIANT]}" == "${BUILD_VARIANT_DRAGONWELL}" ]; then
+  elif [ "${BUILD_CONFIG[BUILD_VARIANT]}" == "${BUILD_VARIANT_DRAGONWELL}" ]; then
+    local dragonwellVerFile=${BUILD_CONFIG[WORKSPACE_DIR]}/${BUILD_CONFIG[WORKING_DIR]}/${BUILD_CONFIG[OPENJDK_SOURCE_DIR]}/version.txt
```
[the remark this answers is not in the document]
Comment (joeyleeeeeee97) [reply sbin/build.sh:153]: Not now, I will add them files soon.
I think Correto and Openj9 using a version file is a better practice, so I want to add that file to Dragonwell.

[the remark this answers is not in the document]
Comment (joeyleeeeeee97) [reply sbin/build.sh:153]: https://github.com/alibaba/dragonwell8/pull/185
https://github.com/alibaba/dragonwell11/pull/53

File: sbin/build.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (andrew-m-leonard) [reply sbin/build.sh:837]: So i've just tried a quick sort check, and I think it does get away with producing the right tag, although strictly speaking the sort is not quite right as the numeric sort fields for the 2nd sort fields 4,4n and 5,5n have characters in them but I believe do produce the correct result. Try:
```
git tag -l "dragonwell-*_jdk*" | grep -v _openj9 | grep -v _adopt | sed 's/jdk-/jdk./g' | sed 's/+/.0.+/g' | sort -t+ -k2,2n | nl | sort -t. -k2,2n -k3,3n -k4,4n -k5,5n | sed 's/\.0\.+/+/g' | cut -f2- | sed 's/jdk./jdk-/g' | tail -1
```
The end sort sequence is:
```
dragonwell-11.0.8.3_jdk-11.0.8-ga
dragonwell-11.0.8.3_jdk-11.0.8+9
dragonwell-11.0.9.3_jdk-11.0.9-ga
dragonwell-11.0.9.3_jdk-11.0.9+0
dragonwell-11.0.9.4_jdk-11.0.9-ga
dragonwell-11.0.9.4_jdk-11.0.9+0
```
As you can see the fields separated by ".", fields 4,4n and 5,5n are, eg:
4,4n = "4_jdk-11"
5,5n = "0"
but I think the numeric sort on these probably produces the right answer:
```
dragonwell-11.0.9.4_jdk-11.0.9+0
```

**Review by andrew-m-leonard — APPROVED** [review]
Looks good now I reckon

**Review by M-Davies — APPROVED** [review]
Looks ok to merge post release

**Review by karianna — APPROVED** [review]
lgtm

**Pull request merged**